internal/pkg/service/stream/keboolasink/bridge/slice.go

```diff
@@ -15,50 +15,50 @@ import (
 func (b *Bridge) uploadSlice(
 	ctx context.Context,
 	volume *diskreader.Volume,
-	slice *model.Slice,
+	slice *plugin.Slice,
 	stats statistics.Value,
 ) error {
 	start := time.Now()
-	reader, err := volume.OpenReader(slice)
+	reader, err := volume.OpenReader(slice.SliceKey, slice.LocalStorage, slice.EncodingCompression, slice.StagingStorage.Compression)
 	if err != nil {
 		b.logger.Warnf(ctx, "unable to open reader: %v", err)
 		return err
 	}
 
+	// Get authorization token
 	token, err := b.schema.Token().ForSink(slice.SinkKey).GetOrErr(b.client).Do(ctx).ResultOrErr()
 	if err != nil {
 		return err
 	}
 
 	defer func() {
-		err = reader.Close(ctx)
+		err := reader.Close(ctx)
 		if err != nil {
-			err = &plugin.ReaderClosureError{SliceKey: slice.SliceKey}
 			b.logger.Warnf(ctx, "unable to close reader: %v", err)
 			return
 		}
 
 		ctx, cancel := context.WithTimeout(ctx, uploadEventSendTimeout)
-		err = b.SendSliceUploadEvent(ctx, b.publicAPI.WithToken(token.String()), time.Since(start), &err, slice, stats)
+		err = b.SendSliceUploadEvent(ctx, b.publicAPI.WithToken(token.String()), time.Since(start), &err, slice.SliceKey, stats)
 		cancel()
 		if err != nil {
-			err = &plugin.SendSliceUploadEventError{SliceKey: slice.SliceKey}
 			b.logger.Warnf(ctx, "unable to send slice upload event: %v", err)
 			return
```

Comment:
This is not supposed to end up with error. Return `nil` and write that we set it to `nil` due to successfull upload and we expect to switch the state of slice to `Uploaded` state even though the event was not dispatched.